java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S1860_java.html

```diff
@@ -1,13 +1,23 @@
 <p>Objects which are pooled and potentially reused should not be used for synchronization. If they are, it can cause unrelated threads to deadlock
 with unhelpful stacktraces. Specifically, <code>String</code> literals, and boxed primitives such as Integers should not be used as lock objects
-because they are pooled and reused. The story is even worse for <code>Boolean</code> objects, because there are only two instances of
-<code>Boolean</code>, <code>Boolean.TRUE</code> and <code>Boolean.FALSE</code> and every class that uses a Boolean will be referring to one of the
-two.</p>
+because they are pooled and reused.</p>
+<p>Here is the list of types which shouldn't be used for synchronization:</p>
+<ul>
+  <li> The primitive wrapper classes in java.lang; </li>
+  <li> The class java.lang.Runtime.Version; </li>
+  <li> The "optional" classes in java.util: Optional, OptionalInt, OptionalLong, and OptionalDouble; </li>
+  <li> Many classes in the java.time API: Instant, LocalDate, LocalTime, LocalDateTime, ZonedDateTime, ZoneId, OffsetTime, OffsetDateTime, ZoneOffset,
+  Duration, Period, Year, YearMonth, and MonthDay, and, in java.time.chrono: MinguoDate, HijrahDate, JapaneseDate, and ThaiBuddhistDate; </li>
+  <li> The interface java.lang.ProcessHandle and its implementation classes; </li>
+  <li> The implementation classes of the collection factories in java.util: List.of, List.copyOf, Set.of, Set.copyOf, Map.of, Map.copyOf,
+  Map.ofEntries, and Map.entry. </li>
+</ul>
```

Comment:
I feel it would be valuable to mention JEP-390 in the description, or in the `See` section.